### PR TITLE
Bug fixes and improvements to `analyse_package`.

### LINKED ISSUE #2: Improve function `analyse_package`
# Potential improvements
- [ ] Add argument `ignore_non_R` to ignore for example .txt files.
- [x] Add argument `is_package` to specify if the directory is the root of a packge, or the R folder.
- [ ] Add argument `language` to specify the programming language (so that the function knows how functions are defined).
- [x] Check that directory argument actually is a directory (like "C:/...").
- [ ] Make the dataframe of functions a tibble instead.
- [ ] Make the unit tests be able to test `analyse_package` on "test functions" with weird variable names in a dedicated file, like "tests/custom_test_material/custom_test_functions.R".
- [x] Add ability to identify one line functions (such as `sayhello <- function(){cat("Hello world")}`).
- [ ] Extend `analyse_package` to accept letters beyond "a-z" and "A-Z" in function names.
- [ ] Extend `analyse_package` to accept function definitions like:
```
abc <-
function
(a,
                 b,
                 c=3){
  cat("It works.\n")
}
```
- [ ] Also add ability to identify functions with function definitions that span multiple lines, like:
```
example <- function(a = 123,
b = "defaultchar",
c = 567)}
```

### PR DESCRIPTION
**Pull request by TheLaughingDuck**

### CODE REVIEW

**Comment by TheLaughingDuck** [comment]
**Current status**: the last commit visible above (accidentally called "Added") did indeed manage to identify one-line functions, but the way this was achieved was through looking for lines that end with left braces ("}"). This in turn introduced a bug where a function would be considered to be ended as soon as a line within it ends with "}". Clearly, there may be many lines ending with "}" within a function (from if/else/while/for etc statements), and thus such functions are condsidered to be ended prematurely, which makes the line count for these functions incorrect.

Attempts to solve this while retaining the ability to identify one-line functions have so far been unsuccessful and have involved long and intricate regex's.

TLDR: Commit "Added" introduced a bug where `analyse_package` thinks a function it is analysing has ended before it actually ends, which makes the line count for functions inaccurate.
